File: samples/sample01/src/main/java/net/jworkflow/sample01/HelloWorkflow.java
package net.jworkflow.sample01;

import net.jworkflow.sample01.steps.Goodbye;
import net.jworkflow.sample01.steps.Hello;
import net.jworkflow.kernel.interfaces.*;

public class HelloWorkflow implements Workflow {

    @Override
    public String getId() {
        return "hello";
    }
    
    @Override
    public int getVersion() {
        return 1;
    }
    
    @Override
    public Class getDataType() {
        return Object.class;
    }

    @Override
    public void build(WorkflowBuilder builder) {
        builder
            .startsWith(Hello.class)
            .then(Goodbye.class);        
    }
}

File: sample01-kotlin/src/main/java/net/jworkflow/sample01/HelloWorkflow.kt
package net.jworkflow.sample01
import net.jworkflow.sample01.steps.Goodbye
import net.jworkflow.sample01.steps.Hello
import net.jworkflow.kernel.interfaces.*

class HelloWorkflow:Workflow {
    val id:String?
        @Override
        get() = "hello"

    val version:Int
        @Override
        get() = 1

    val dataType:Class?
        @Override
        get() = Object::class.java

    @Override
    fun build(builder:WorkflowBuilder?) {
        builder!!
                .startsWith(Hello::class.java)
                .then(Goodbye::class.java)
    }
}